src/lib.rs

```diff
@@ -20,13 +20,19 @@ pub enum WeightUnit {
 }
 
 fn setup_store<T>(store: &mut Store<T>, weight_unit: WeightUnit) {
+    let fuel_max: u64 = store
```

Comment:
what I would do is that if the weight unit is `Fuel` then we set the fuel to `u64::MAX`

Reply:
>but what we could do is reset it at every step to u64::MAX

That would break function runner- it inspects the fuel usage at the end of the process and uses it to determine how many instructions were used. 

>so that it should never go to 0.

For an arbitrary client, they might want their system to time out after a certain amount of fuel is used, this would break that in a somewhat confusing way by mutating the amount of fuel while setting up the profiler

Reply:
I don't think we run the risk of using `u64::max` fuel, no? Based on some naive calculations that's ~1,700 years, assuming ~11M fuel is roughly 5ms.

Maybe an alternative approach would be to set the fuel to `u64::max` only if the store didn't have fuel.

Reply:
>I don't think we run the risk of using u64::max fuel, no? Based on some naive calculations that's ~1,700 years, assuming ~11M fuel is roughly 5ms.

Yes, certainly once it's been set to u64::max, it won't time out. Which could be an issue if the client wanted it to timeout, or otherwise was tracking the fuel itself.

>Maybe an alternative approach would be to set the fuel to u64::max only if the store didn't have fuel.

This resolves the issue of overriding a carefully set fuel limit, but still runs into ordering issues:
- Client sets up profiler, profiler sets fuel to u64::max
- Client then sets up a non-trivial fuel limit, say 11 million
- The profiler callback executes- the first instruction appears to have consumed `u64::max - 11 million` instructions

My code here doesn't completely avoid this as there might be other reasons why the client would change the fuel value mid execution, but it does prevent the case in which the client simply reverses the ordering of the profiler initialization and fuel initialization by raising an error

Reply:
In this case, they might be adding fuel after the profiler is setup, but before wasmtime executes, so we can't rely on wasmtime behaviour to notify them of the issue 🤔